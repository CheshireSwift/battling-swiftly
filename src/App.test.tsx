import * as React from 'react'
import { render, mount } from 'enzyme'

import App from './App'
import { act } from 'react-dom/test-utils'

describe('the app', () => {
  const docs = [
    {
      id: 'hero',
      data: () => ({
        color: 'lime',
        name: 'Waffle',
        position: {
          x: 257.99202542018276,
          y: 636.24685445642828,
        },
      }),
    },
    {
      id: 'villain',
      data: () => ({
        color: 'orange',
        name: 'Pickle',
        position: {
          x: 457.99202542018276,
          y: 236.24685445642828,
        },
      }),
    },
  ]

  const buildStore = (promise = Promise.resolve({ docs })) => ({
    collection: () => ({
      get: () => promise,
      onSnapshot: jest.fn(),
      doc: jest.fn(),
    }),
  })

  it('matches the snapshot', async () => {
    jsdom.reconfigure({
      url:
        'http://example.com/?bg=some_image.png&dpi=30&collection=test-collection&char=hero',
    })

    const promise = Promise.resolve({ docs })
    const store = buildStore(promise)
    const app = mount(
      <div>
        <App store={store} />
      </div>,
    )
    await act((async () => {
      await promise
      app.update()
    }) as any)

    expect(app.html()).toMatchSnapshot()
  })

  it('matches the snapshot with bg missing', () => {
    jsdom.reconfigure({
      url: 'http://example.com/?dpi=30&collection=test-collection&char=hero',
    })

    expect(render(<App store={buildStore()} />)).toMatchSnapshot()
  })

  it('matches the snapshot with dpi missing', () => {
    jsdom.reconfigure({
      url:
        'http://example.com/?bg=some_image.png&collection=test-collection&char=hero',
    })

    expect(render(<App store={buildStore()} />)).toMatchSnapshot()
  })

  it('matches the snapshot with collection missing', () => {
    jsdom.reconfigure({
      url: 'http://example.com/?bg=some_image.png&dpi=30&char=hero',
    })

    expect(render(<App store={buildStore()} />)).toMatchSnapshot()
  })

  it('matches the snapshot with multiple controlled characters', async () => {
    jsdom.reconfigure({
      url:
        'http://example.com/?bg=some_image.png&dpi=30&collection=test-collection' +
        '&char=hero' +
        '&char=villain',
    })

    const promise = Promise.resolve({ docs })
    const store = buildStore(promise)
    const app = mount(
      <div>
        <App store={store} />
      </div>,
    )
    await act((async () => {
      await promise
      app.update()
    }) as any)

    expect(app.html()).toMatchSnapshot()
  })
})
