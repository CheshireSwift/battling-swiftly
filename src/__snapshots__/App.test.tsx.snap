// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the app matches the snapshot 1`] = `
"<div><img style=\\"position: absolute; top: 0px; bottom: 0px;\\" src=\\"some_image.png\\"><div><svg style=\\"position: absolute; top: 0px; bottom: 0px; height: 0px; width: 0px;\\"><circle cx=\\"257.99202542018276\\" cy=\\"636.2468544564283\\" r=\\"4\\" stroke=\\"lime\\" stroke-width=\\"1.5\\" fill=\\"lime\\" fill-opacity=\\"0.1\\"></circle><path stroke=\\"lime\\" stroke-width=\\"1.5\\" fill=\\"none\\" d=\\"
M 259.99202542018276,634.2468544564283
L 272.99202542018276,623.2468544564283
L 308.99202542018276,623.2468544564283
    \\"></path><text x=\\"272.99202542018276\\" y=\\"621.2468544564283\\" fill=\\"lime\\" style=\\"font-family: monospace; font-weight: bold; font-size: 14px;\\">Waffle</text><text x=\\"272.99202542018276\\" y=\\"621.2468544564283\\" fill=\\"lime\\" style=\\"font-family: monospace; font-size: 14px;\\"></text><circle cx=\\"457.99202542018276\\" cy=\\"236.24685445642828\\" r=\\"4\\" stroke=\\"orange\\" stroke-width=\\"1.5\\" fill=\\"orange\\" fill-opacity=\\"0.1\\"></circle><path stroke=\\"orange\\" stroke-width=\\"1.5\\" fill=\\"none\\" d=\\"
M 459.99202542018276,234.24685445642828
L 472.99202542018276,223.24685445642828
L 508.99202542018276,223.24685445642828
    \\"></path><text x=\\"472.99202542018276\\" y=\\"221.24685445642828\\" fill=\\"orange\\" style=\\"font-family: monospace; font-weight: bold; font-size: 14px;\\">Pickle</text><text x=\\"472.99202542018276\\" y=\\"221.24685445642828\\" fill=\\"orange\\" style=\\"font-family: monospace; font-size: 14px;\\"></text></svg></div></div>"
`;

exports[`the app matches the snapshot with bg missing 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "Require background image (?bg=url)",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`the app matches the snapshot with collection missing 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "Require data collection (?collection=collection-name)",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`the app matches the snapshot with dpi missing 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "Require dpi (?dpi=number)",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`the app matches the snapshot with multiple controlled characters 1`] = `
"<div><img style=\\"position: absolute; top: 0px; bottom: 0px;\\" src=\\"some_image.png\\"><div><svg style=\\"position: absolute; top: 0px; bottom: 0px; height: 0px; width: 0px;\\"><circle cx=\\"257.99202542018276\\" cy=\\"636.2468544564283\\" r=\\"4\\" stroke=\\"lime\\" stroke-width=\\"1.5\\" fill=\\"lime\\" fill-opacity=\\"0.1\\"></circle><path stroke=\\"lime\\" stroke-width=\\"1.5\\" fill=\\"none\\" d=\\"
M 259.99202542018276,634.2468544564283
L 272.99202542018276,623.2468544564283
L 308.99202542018276,623.2468544564283
    \\"></path><text x=\\"272.99202542018276\\" y=\\"621.2468544564283\\" fill=\\"lime\\" style=\\"font-family: monospace; font-weight: bold; font-size: 14px;\\">Waffle</text><text x=\\"272.99202542018276\\" y=\\"621.2468544564283\\" fill=\\"lime\\" style=\\"font-family: monospace; font-size: 14px;\\"></text><circle cx=\\"457.99202542018276\\" cy=\\"236.24685445642828\\" r=\\"4\\" stroke=\\"orange\\" stroke-width=\\"1.5\\" fill=\\"orange\\" fill-opacity=\\"0.1\\"></circle><path stroke=\\"orange\\" stroke-width=\\"1.5\\" fill=\\"none\\" d=\\"
M 459.99202542018276,234.24685445642828
L 472.99202542018276,223.24685445642828
L 508.99202542018276,223.24685445642828
    \\"></path><text x=\\"472.99202542018276\\" y=\\"221.24685445642828\\" fill=\\"orange\\" style=\\"font-family: monospace; font-weight: bold; font-size: 14px;\\">Pickle</text><text x=\\"472.99202542018276\\" y=\\"221.24685445642828\\" fill=\\"orange\\" style=\\"font-family: monospace; font-size: 14px;\\"></text></svg></div><div style=\\"position: fixed; background: black; color: lime; top: 0px; right: 0px;\\"><div id=\\"character-selector-option-hero\\" style=\\"padding: 0.5rem; border: 1px solid lime;\\">Waffle</div><div id=\\"character-selector-option-villain\\" style=\\"padding: 0.5rem; border: 1px solid black;\\">Pickle</div></div></div>"
`;
